lib/matplotlib/axis.py

```diff
@@ -1023,6 +1023,97 @@ def set_default_intervals(self):
         # attribute, and the derived code below will check for that
         # and use it if it's available (else just use 0..1)
 
+    def _set_lim(self, v0, v1, alt0=None, alt1=None, *,
+                 emit=True, auto, names=("", "")):
+        """
+        Set view limits.
+
+        This method is a helper for the Axes ``set_xlim``, ``set_ylim``, and
+        ``set_zlim`` methods.  This docstring uses names corresponding to
+        ``set_xlim`` for simplicity.
+
+        *names* is the pair of the names of the first two parameters of the
+        Axes method (e.g., "left" and "right").  They are only used to generate
+        error messages; and can be empty if the limits are known to be valid.
+
+        Other parameters are directly forwarded from the Axes limits setter:
+        *v0*, *v1*, *alt0*, and *alt1* map to *left*, *right*, *xmin*, and
+        *xmax* respectively; *emit* and *auto* are used as is.
+        """
+        v0name, v1name = names  # The value names.
+        name, = [name for name, axis in self.axes._get_axis_map().items()
+                 if axis is self]  # The axis name.
+
+        if v1 is None and np.iterable(v0):
+            v0, v1 = v0
+        if alt0 is not None:
+            if v0 is not None:
+                raise TypeError(
+                    f"Cannot pass both {v0name!r} and '{name}lim'")
+            v0 = alt0
+        if alt1 is not None:
+            if v1 is not None:
+                raise TypeError(
+                    f"Cannot pass both {v1name!r} and '{name}lim'")
+            v1 = alt1
```

Comment:
I'm wondering whether this must be part of an internal `_set_lim()` function. This is only needed to support some weird parameter duplication in the calling functions. Can we leave that part in the calling functions? That would keep it more readable the variable mapping is hard to follow. Alternatively, could we have a decorator

```
@_api.alias(left='xlim', right='ylim')
def set_xlim(self, left=None, right=None, emit=True, auto=False):
```
that does the checking? Then, the aliases don't show in the signature, but I would consider that a plus. IMHO it's better to have the signature only illustrate one case, and use the docstring to say "You can use 'xmin' instead of 'left' and ...". That's a bit peculiar but then again, the duplication is quite confusing as well.

Reply:
I can move back the parameter dealiasing to set_{x,y,z}lim and duplicate them 3x again, but I'm not sure if this really helps...  (In particular, you still have to pass `names`, which are used further down as well)  Also, moving that logic to a separate `_api.alias` also seems more complexity than needed (in particular by splitting the logic to two different functions/files).

I do agree that the signature can be clearer, but that can be done with `set_xlim.__signature__ = inspect.signature(lambda self, left=None, right=None, emit=True, auto=False: None)` (which is probably a good idea, but can be done separately).

Reply:
> I can move back the parameter dealiasing to set_{x,y,z}lim and duplicate them 3x again, but I'm not sure if this really helps...

I think yes. It's some structural reduncancy, but the names are always different and having the original names and the context of the docstring helps.

Additionally, it's a great simplification if `set_lim` only has to bother with two values for the limits instead of four. Even though two of them are only used at the top, the function is so long that you can see that directly.

You may even get away without the names by re-wording a little. See below.

Reply:
OK, done (as a separate commit).

The "canonical" property names for Axes3D were switched from xlim3d/ylim3d to xlim/ylim so that Axes3D can directly inherit the setters from 2D Axes (avoiding more duplication there).